shopping_mall/backend/app/core/config.py

```
@@ -52,6 +57,21 @@ class Settings(BaseSettings):
+    def langgraph_postgres_url(self) -> str:
+        """AsyncPostgresSaver(psycopg3)용 URL — SQLAlchemy 드라이버 접미사 제거.
+
+        예: postgresql+asyncpg://... → postgresql://...
+        PostgreSQL URL이 아닌 경우 ValueError를 발생시킵니다.
+        """
+        url = self.database_url
+        if not re.match(r"^postgres(?:ql)?(?:\+\w+)?://", url):
+            raise ValueError(
+                f"DATABASE_URL이 PostgreSQL URL이 아닙니다: {url!r}. "
+                "AsyncPostgresSaver는 PostgreSQL 전용입니다."
+            )
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)